File: spark-in-action/java/src/main/java/ingestion/file/json/JsonLinesToDataframeApp.java
package ingestion.file.json;

import org.apache.spark.sql.Dataset;
import org.apache.spark.sql.Row;
import org.apache.spark.sql.SparkSession;

public class JsonLinesToDataframeApp {
    public static void main(String[] args) {
        JsonLinesToDataframeApp app = new JsonLinesToDataframeApp();
        app.start();
    }

    private void start() {
        SparkSession spark = SparkSession.builder()
                .appName("JSON Lines to Dataframe")
                .master("local")
                .getOrCreate();

        spark.sparkContext().setLogLevel("ERROR");

        // read a json file
        Dataset<Row> df = spark.read().format("json")
                .load("data/ch07/durham-nc-foreclosure-2006-2016.json");

        df.show(5);
        df.printSchema();
    }
}

File: spark-in-action/scala/src/main/scala/ingestion/file/json/JsonLinesToDataframeApp.scala
package ingestion.file.json

import org.apache.spark.sql.SparkSession

object JsonLinesToDataframeApp {

  def main(args: Array[String]): Unit = {
    // create a session on a local master
    val spark = SparkSession.builder
      .appName("Json Lines to Dataframe")
      .master("local[*]")
      .getOrCreate

    spark.sparkContext.setLogLevel("ERROR")

    val df = spark.read.format("json")
      .load("data/ch07/durham-nc-foreclosure-2006-2016.json")
    
    df.show(5)
    df.printSchema

    spark.stop
  }

}
